plugins/logs/plugin.go

```diff
@@ -553,6 +568,27 @@ func (p *Plugin) Reconfigure(_ context.Context, config interface{}) {
 	<-done
 }
 
+// Trigger can be used to control when the plugin attempts to upload
+// a new decision log in manual triggering mode.
+func (p *Plugin) Trigger(ctx context.Context) {
+	done := make(chan struct{})
+	go func() {
+		if p.config.Service != "" {
+			_ = p.doOneShot(ctx)
+		}
+
+		if ctx.Err() != nil {
+			return
+		}
```

Comment:
🤔 Does this actually matter? In the `select` that follows, the same thing would happen, regardless of whether we've called `close(done)` or not.

Reply:
If we get a context error, the ctx.Done() channel would close and the select loop would break. I thought calling `close(done)` in that case would block since no one is listening on the `done` channel. It's likely I'm wrong on that.
I've removed that check.

Reply:
Closing channels is fine, I think, it won't be blocking. Sends on closed channels panic, receives in closed channels return the nil value immediately. So we should be OK. Thanks!